### ISSUE TITLE
[Module Proposal]: `avm-res-monitor-diagnostic-settings`

### ISSUE THREAD

**Issue by jrspinella** [Azure/Azure-Verified-Modules#918]
### Check for previous/existing GitHub issues/module proposals

- [X] I have checked for previous/existing GitHub issues/module proposals.

### Check this module doesn't already exist in the module indexes

- [X] I have checked for that this module doesn't already exist in the module indexes; or I'm proposing the module to be migrated from CARML/TFVM.

### Bicep or Terraform?

Terraform

### Module Classification?

Resource Module

### Module Name

avm-res-monitor-diagnostic-settings

### Module Details

This module can create Diagnostic Settings on a given Azure resource and manage related parameters.

### Do you want to be the owner of this module?

Yes

### Module Owner's GitHub Username (handle)

jrspinella

**Comment by mathiasjespers** [comment]
> @jrspinella diagnostic settings configuration is mandatory AVM module interface. Why you think that we should have this as separated module?

This is indeed debatable, but a lot of people like to create a seperate module for storage account and use that module instead of configuring it in each resource module. Call it a preference but this is is defininetly something people do

**Comment by prjelesi** [comment]
Agree. Let's keep this open and make it as request to collect feedback from field. Tnx @mathiasjespers

**Comment by shghildi** [comment]
Can u pls help me understand how to contribute for this module ?

**Comment by chinthakaru** [comment]
We don't see value in this module based on the effort and impact. Therefore, we are closing this module due to inactivity. If anyone feels that this needs to be addressed, please feel free to reopen the issue.

cc @matt-FFFFFF

**Comment by EAR009** [comment]
I'd like to request reconsideration of the avm-res-monitor-diagnostic-settings proposal (#918), which was initially closed as "Invalid." Since April 2024, we’ve seen strong organizational value in modularizing diagnostic settings because:

• It prevents repetition and inconsistencies across resource modules (DRY principle).
• Maintenance becomes far easier—schema updates, new destinations, policy tweaks happen centrally.
• It enhances reusable governance controls, ensuring uniform diagnostics across all resources.
• It aligns with infrastructure modularity principles like single-responsibility and composability (CUPID).
• AVM specs (e.g., RMFR2/RMFR4) don’t prohibit this; they only require optional diagnostics support—implementation methods are flexible.

Can the maintainers please clarify the reasons for marking the proposal invalid? If the concern was about AVM guidance evolving, perhaps we can revisit with a refined proposal that aligns better with current standards and addresses maintainability and reusability concerns.
